apollo-api/src/main/java/com/apollographql/apollo/api/Operation.java

```diff
@@ -5,28 +5,60 @@
 
 import javax.annotation.Nonnull;
 
-/** TODO */
+/**
+ * Represents an abstraction for a GraphQL operation (mutation or query).
+ */
 public interface Operation<D extends Operation.Data, T, V extends Operation.Variables> {
-  /** TODO */
+  /**
+   * Returns the raw GraphQL operation string.
+   */
   String queryDocument();
 
-  /** TODO */
+  /**
+   * Returns the variables associated with this GraphQL operation.
+   */
   V variables();
 
+  /**
+   * Returns a mapper that maps the server response data to generated model class {@link D}.
+   */
   ResponseFieldMapper<D> responseFieldMapper();
 
+  /**
+   * Wraps the generated response data class {@link D} with another class. For example, a use case for this would be to
+   * wrap the generated response data class in an Optional i.e. Optional.fromNullable(data).
+   */
   T wrapData(D data);
 
-  /** TODO */
+  /**
+   * Abstraction for data returned by the server in response to this operation.
+   */
   interface Data {
   }
 
-  /** TODO */
+  /**
+   * Abstraction for the variables which are a part of the GraphQL operation. For example, for the following GraphQL
+   * operation, Variables represents values for GraphQL '$type' and '$limit' variables:
+   * <pre>{@code
+   *      query FeedQuery($type: FeedType!, $limit: Int!) {
+   *          feedEntries: feed(type: $type, limit: $limit) {
+   *          id
+   *          repository {
+   *              ...RepositoryFragment
+   *          }
+   *          postedBy {
+   *            login
+   *          }
+   *      }
+   *    }
+   * }
+   * </pre>
+   */
   class Variables {
     protected Variables() {
     }
 
-    @Nonnull protected Map<String, Object> valueMap() {
+    @Nonnull public Map<String, Object> valueMap() {
```

Comment:
We moved the Field class to the internal package (which was earlier in the same package as the Operation interface) and Field class was using valueMap method. Hence it had to be changed to public.